config/src/cluster_config.rs

```diff
@@ -34,20 +34,20 @@ impl ClusterConfig {
         vec!["127.0.0.1:8901".to_string()]
     }
 
-    fn default_http_listen_port() -> u16 {
-        8902
+    fn default_http_listen_port() -> Option<u16> {
```

Comment:
As you removed serde derive **default**, these functions `default_xxx_listen_port()` are useless. If you want to keep these default listened port in code, I have an idea but the config pattern needs to be modified:

```toml
[cluster]

[cluster.http]
enabled = true
listen_port = 8902

[cluster.grpc]
enabled = true
listen_port = 8903


```